src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```
@@ -202,7 +218,7 @@ private Object createObjectWithIgnoringProblems(String className, Set<String> se
      * @return a string which is obtained by joining package names with a class name.
      */
     private static String joinPackageNamesWithClassName(String className, Set<String> packages) {
-        return packages.stream().filter(name -> name != null)
+        return packages.stream().filter(Objects::nonNull)
```

I do not know reason.
We could raise an exception if user provide us nulls.